clang/lib/Sema/SemaType.cpp

```diff
@@ -7024,13 +7024,12 @@ static bool handleFunctionTypeAttr(TypeProcessingState &state, ParsedAttr &attr,
                << (int)Sema::CallingConventionIgnoredReason::VariadicFunction;
 
       attr.setInvalid();
-      if (S.getLangOpts().SYCLIsDevice) {
-        S.SYCLDiagIfDeviceCode(attr.getLoc(), diag::err_cconv_varargs)
-          << FunctionType::getNameForCallConv(CC);
-        return true;
-      } else
-      return S.Diag(attr.getLoc(), diag::err_cconv_varargs)
-             << FunctionType::getNameForCallConv(CC);
+      // Disable this diagnostic for SYCL
```

Comment:
SYCL 1.2.1 specification explicitly prohibits variadic functions:
"No variadic functions are allowed to be called in a SYCL kernel or any functions called by the kernel."

NOTE: variadic functions are allowed in the "host code".

Reply:
Hi Alexey,

I want to issue this diagnostic at the call site - i.e. when such a function is called in the kernel context.   Currently the diagnostic is issued when the declaration is simply seen.   That seems wrong and that is what I have disabled.